modules/library/main/src/main/java/org/geotools/data/AbstractDataAccessFactory.java

```diff
@@ -0,0 +1,189 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2012, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.geotools.data.DataAccessFactory.Param;
+import org.geotools.parameter.DefaultParameterDescriptor;
+import org.geotools.parameter.DefaultParameterDescriptorGroup;
+import org.geotools.parameter.FloatParameter;
+import org.geotools.parameter.Parameter;
+import org.opengis.parameter.ParameterDescriptorGroup;
+import org.opengis.parameter.ParameterValue;
+
+/**
+ * This implementation was lifted from AbstractDataStoreFactory.
+ * As there was nothing in the implementation that was Store-specific (i.e. a restriction on simple types)
+ * this was raised to allow the factory to be used for complex features.
```

Comment:
That sounds fine, if this class provides value I do not mind it being around. When rewriting the ContentDataStore tutorial we ended setting up a few utility methods (rather than an entire base class).